helix-core/src/main/java/org/apache/helix/task/TaskUtil.java

```diff
@@ -1021,11 +1021,21 @@ public static void purgeExpiredJobs(String workflow, WorkflowConfig workflowConf
           // concurrent write issue. It is possible that jobs got purged but there is no event to
           // trigger the pipeline to clean context.
           HelixDataAccessor accessor = manager.getHelixDataAccessor();
-          List<String> resourceConfigs =
-              accessor.getChildNames(accessor.keyBuilder().resourceConfigs());
+          List<ResourceConfig> resourceConfigs =
```

Comment:
getChildNames is a single ZK call, but getChildValues are fetching the contents of all resource configs. If the volume of resource config is large, the performance impact (latency of this call + additional traffic to ZK) is no-ignorable.  Is that possible to have other way to do this? Or as my thinking, do we really need this touch at all?

Reply:
It will impact the traffic, but it should not impact the latency of pipeline since this is in the asynchronized thread. I think we can revisit it when we are trying to get rid of all "touches".

Reply:
Since you are getting rid of all "touches" anyway, other option is to keep the code as it is now without changing anything.  How do you think?